lua/randomat2/sh_events/sh_poker.lua

```diff
@@ -0,0 +1,258 @@
+ConVars.EnableContinuousPlay = ConVars.EnableContinuousPlay or CreateConVar("randomat_poker_enable_continuous_play", "0", FCVAR_REPLICATED, "Enables continuous play, event repeats until TTT game ends")
+ConVars.EnableSmallerBets = ConVars.EnableSmallerBets or CreateConVar("randomat_poker_enable_smaller_bets", "0", FCVAR_REPLICATED, "Enables smaller bet increments (default: 25-50-75-100, alt: 10-20-30-...-100)")
+ConVars.EnableNineDiamondsGag = ConVars.EnableNineDiamondsGag or CreateConVar("randomat_poker_enable_nine_diamonds", "1", FCVAR_REPLICATED, "Enables the 9 of Diamonds win condition gag")
+
+ConVars.EnableRandomCollusions = ConVars.EnableRandomCollusions or CreateConVar("randomat_poker_colluding_enable_random_collusions", "1", FCVAR_REPLICATED, "Enables whether your colluding partner should be randomized or ordered")
+ConVars.AnonymizeCollusions = ConVars.AnonymizeCollusions or CreateConVar("randomat_poker_colluding_anonymized_collusions", "1", FCVAR_REPLICATED, "Enables whether the colluding partner's name should be hidden")
+
+DynamicTimerPlayerCount = 0
+function GetDynamicRoundTimerValue(conVar)
+    if ConVars.ManualRoundStateTimes:GetBool() then
+        return ConVars[conVar]:GetInt()
+    elseif ({RoundStateMessage = true, RoundStateStart = true, RoundStateEnd = true})[conVar] then
+        return 5
+    else
+        local window = 30
+
+        if DynamicTimerPlayerCount >= 5 then
+            window = 20
+        end
+
+        return window
+    end
+end
+
+
+BettingStatus = {
+    NONE = 0,
+    FOLD = 1,
+    CHECK = 2,
+    CALL = 3,
+    RAISE = 4,
+    ALL_IN = 5
+}
+
+Bets = {
+    NONE = 0,
+    QUARTER = 1,
+    HALF = 2,
+    THREEQ = 3,
+    ALL = 4
+}
+
+Bets_Alt = {
+    NONE = 0,
+    TEN = 1,
+    TWENTY = 2,
+    THIRTY = 3,
+    FOURTY = 4,
+    FIFTY = 5,
+    SIXTY = 6,
+    SEVENTY = 7,
+    EIGHTY = 8,
+    NINETY = 9,
+    ALL = 10
+}
+
+Hands = {
+    NONE = 0,
+    HIGH_CARD = 1,
+    PAIR = 2,
+    TWO_PAIR = 3,
+    THREE_KIND = 4,
+    STRAIGHT = 5,
+    FLUSH = 6,
+    FULL_HOUSE = 7,
+    FOUR_KIND = 8,
+    STRAIGHT_FLUSH = 9,
+    ROYAL_FLUSH = 10,
+    NINE_OF_DIAMONDS = 11
+}
+
+Cards = {
+    NONE = 0,
+    ACE = 1,
+    TWO = 2,
+    THREE = 3,
+    FOUR = 4,
+    FIVE = 5,
+    SIX = 6,
+    SEVEN = 7,
+    EIGHT = 8,
+    NINE = 9,
+    TEN = 10,
+    JACK = 11,
+    QUEEN = 12,
+    KING = 13,
+    ACE_HIGH = 14
+}
+
+Suits = {
+    NONE = 0,
+    SPADES = 1,
+    HEARTS = 2,
+    DIAMONDS = 3,
+    CLUBS = 4
+}
+
+function CardRankToFileName(rank)
+    if rank == Cards.NONE then
+        return ""
+    elseif rank == Cards.ACE then
+        return "ace"
+    elseif rank == Cards.JACK then
+        return "jack"
+    elseif rank == Cards.QUEEN then
+        return "queen"
+    elseif rank == Cards.KING then
+        return "king"
+    else
```

Comment:
This is more of a stylistic thing (so you can ignore it) but you can skip the last `else` and just `return`.
e.g.
```lua
if a then
 return 1
elseif b then
 return 2
end
return 3
```

Reply:
Totally just talking theory here: depending on the context, I may prefer the explicit if-else with each having separate returns when the point of the code block is an explicit split of functionality (if a, return 1, if b, return two, else, return 3). In other scenarios, I may skip defining it as an "else" block if that's the core functionality intended by the block, and the early returns are logically "early escapes", where we're accounting for some pre-existing condition altering the standard functionality in some way.

In other words, this would be a case statement if I wasn't writing Lua code.